packages/@ourworldindata/components/src/IndicatorSources/IndicatorSources.tsx

```diff
@@ -1,109 +1,150 @@
 import React from "react"
+import cx from "classnames"
 import { ExpandableToggle } from "../ExpandableToggle/ExpandableToggle.js"
-import { OwidOrigin, dayjs, uniqBy } from "@ourworldindata/utils"
+import { DisplaySource, dayjs, uniqBy } from "@ourworldindata/utils"
 import { SimpleMarkdownText } from "../SimpleMarkdownText.js"
 import { CodeSnippet } from "../CodeSnippet/CodeSnippet.js"
 import { REUSE_THIS_WORK_SECTION_ID } from "../SharedDataPageConstants.js"
 
-export type OriginSubset = Pick<
-    OwidOrigin,
-    | "title"
-    | "producer"
-    | "descriptionSnapshot"
-    | "dateAccessed"
-    | "urlMain"
-    | "description"
-    | "citationFull"
->
-
 export interface IndicatorSourcesProps {
-    origins: OriginSubset[]
-    canonicalUrl?: string
+    sources: DisplaySource[]
+    isEmbeddedInADataPage?: boolean // true by default
 }
 
 export const IndicatorSources = (props: IndicatorSourcesProps) => {
-    const origins = props.origins.map((origin) => ({
-        ...origin,
-        label: makeLabel(origin),
-    }))
-    const uniqueOrigins = uniqBy(origins, "label")
+    const isEmbeddedInADataPage = props.isEmbeddedInADataPage ?? true
+    const uniqueSources = uniqBy(props.sources, "label")
 
     return (
         <>
-            {uniqueOrigins.map((source, idx: number, sources) => (
-                <ExpandableToggle
-                    key={source.label}
-                    label={source.label}
-                    content={
-                        <SourceContent
-                            source={source}
-                            canonicalUrl={props.canonicalUrl}
-                        />
-                    }
-                    isStacked={idx !== sources.length - 1}
-                    hasTeaser
-                />
-            ))}
+            {uniqueSources.map((source: DisplaySource, idx: number) => {
+                const isStacked = idx !== uniqueSources.length - 1
+                const content = (
+                    <SourceContent
+                        source={source}
+                        isEmbeddedInADataPage={isEmbeddedInADataPage}
+                    />
+                )
+                return source.description ||
+                    source.citation ||
+                    source.dataPublishedBy ? (
+                    <ExpandableToggle
```

Comment:
I wonder if we should show this section expanded if we are in the "About this data" overlay? The data page is pretty long and shows a lot of stuff so the collapsed section makes sense, but if you want to see the sources then why make users click once more? I'm not 100% sure this is the best way forward but this is my hunch.